lib/spotify/responder.ex

```diff
@@ -12,8 +12,29 @@ defmodule Responder do
       def handle_response({ :ok, %HTTPoison.Response{ status_code: code, body: "" }})
         when code in 200..299, do: :ok
 
-      def handle_response({ message, %HTTPoison.Response{ status_code: code, body: body }})
-        when code in 400..499, do: { message, Poison.decode!(body)}
+      # special handling for 'too many requests' status
+      # in order to know when to retry
+      def handle_response(
+        { message, %HTTPoison.Response{ status_code: 429, body: body, headers: headers }}
+      ) do
+        {retry_after, ""} = (
+          headers
+          |> Enum.find(&(Kernel.elem(&1, 0) == "Retry-After"))
+          |> Kernel.elem(1)
+          |> Integer.parse
+        )
+
+        { message,
+          Map.put(Poison.decode!(body), "meta", %{ "retry_after" => retry_after })
```

Comment:
I wasn't quite sure the best way to include this data in the response.  Creating a `meta` key in the response `Map` was the best I could think of -- open to other ideas though!